File: spring-boot-starter-monitor/src/main/java/org/toolkit/spring/boot/starter/monitor/interceptors/AccessInterceptor.java
package org.toolkit.spring.boot.starter.monitor.interceptors;

import cn.hutool.http.useragent.UserAgentParser;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import java.util.Objects;
import lombok.extern.slf4j.Slf4j;
import lombok.val;
import org.jetbrains.annotations.NotNull;
import org.springframework.web.servlet.HandlerInterceptor;
import org.springframework.web.servlet.ModelAndView;

@Slf4j
public class AccessInterceptor implements HandlerInterceptor {

	private static final String USER_AGENT = "user-agent";

	@Override
	public void postHandle(
			@NotNull HttpServletRequest request,
			@NotNull HttpServletResponse response,
			@NotNull Object handler,
			ModelAndView modelAndView) {
		val agentText = request.getHeader(USER_AGENT);
		if (Objects.isNull(agentText) || agentText.equals(agentText.intern())) return;
		val agent = UserAgentParser.parse(agentText);
	}
}

File: website/src/main/kotlin/org/toolkit/spring/boot/website/interceptor/AccessInterceptor.kt
package org.toolkit.spring.boot.website.interceptor

import jakarta.annotation.Resource
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.springframework.stereotype.Component
import org.springframework.web.servlet.HandlerInterceptor
import org.toolkit.spring.boot.utils.kotlin.annotation.KotlinSlf4j
import org.toolkit.spring.boot.website.entity.AccessEntity
import org.toolkit.spring.boot.website.repository.AccessEntityRepository
import java.lang.Exception

@Component
@KotlinSlf4j
class AccessInterceptor : HandlerInterceptor {
    @Resource
    private lateinit var accessEntityRepository: AccessEntityRepository

    override fun afterCompletion(
        request: HttpServletRequest,
        response: HttpServletResponse,
        handler: Any,
        ex: Exception?,
    ) {
        if (ex != null) return
        val userAnent = request.getHeader("User-Agent")
        val ipAddress = request.remoteHost
        val port = request.remotePort
        val entity = AccessEntity(userAnent, ipAddress, port)
        accessEntityRepository.save(entity)
    }
}
